rhino/src/test/java/org/mozilla/javascript/SlotMapTest.java

```diff
@@ -51,11 +53,11 @@ public void crudOneString() {
         assertEquals(1, map.size());
         assertFalse(map.isEmpty());
         Slot newSlot = new Slot(slot);
-        map.replace(slot, newSlot);
+        map.compute("foo", 0, (k, i, e) -> newSlot);
         Slot foundNewSlot = map.query("foo", 0);
         assertEquals("Testing", foundNewSlot.value);
         assertSame(foundNewSlot, newSlot);
-        map.remove("foo", 0);
+        map.compute("foo", 0, (k, ii, e) -> null);
```

Comment:
Sorry, but for me this code looks absolutely unreadable - i guess nobody is able to understand what this line of code does. It might be technically clever to do it like this but from a maintainers point of view i need more support.

Please think (again) about better method naming and at least add some jdoc to this magic method.

Reply:
I never fully grasped how the Slots and SlotMaps worked and was wondering when I saw this PR coming by whether we could have some blogpost-style explainer of it, either in the Wiki or in the documentation